File: apps/pos/native-windows/DesktopKitchenPOS/Models/Employee.cs
using System.Text.Json.Serialization;

namespace DesktopKitchenPOS.Models;

public record Employee
{
    [JsonPropertyName("id")]
    public int Id { get; init; }

    [JsonPropertyName("name")]
    public string Name { get; init; } = "";

    [JsonPropertyName("pin")]
    public string? Pin { get; init; }

    [JsonPropertyName("role")]
    public string Role { get; init; } = "cashier";

    [JsonPropertyName("active")]
    public bool Active { get; init; } = true;

    [JsonPropertyName("created_at")]
    public string? CreatedAt { get; init; }

    [JsonPropertyName("token")]
    public string? Token { get; init; }

    public string DisplayRole => char.ToUpper(Role[0]) + Role[1..];
}

public record LoginRequest
{
    [JsonPropertyName("pin")]
    public string Pin { get; init; } = "";
}

File: apps/pos/native-android/app/src/main/java/com/desktopkitchen/pos/models/Employee.kt
package com.desktopkitchen.pos.models

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Employee(
    val id: Int,
    val name: String,
    val pin: String? = null,
    val role: String,
    val active: Boolean = true,
    val created_at: String? = null,
    val token: String? = null
) {
    val displayRole: String get() = role.replaceFirstChar { it.uppercase() }
}
